src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,112 @@
+        }
+    }
+
+    public static String[] decodeInput(String input) {
+        return input.split(" ");
+    }
+
+    public static void crossOff(String[] inputWords) {
+        if (!Validation.isValidCrossOff(inputWords)) {
+            DukeException.invalidCrossOffException();
+            return;
+        }
+        int taskIndex = Integer.parseInt(inputWords[1]) - 1;
+        if (!Validation.isValidTaskIndex(taskIndex, taskCount)) {
+            DukeException.invalidTaskIndexException();
+            return;
+        }
+        Task current = tasks.get(taskIndex);
+        current.setDone();
+        UI.printDoneMessage(current);
+    }
+
+    public static void deleteTask(String[] inputWords) {
+        if (!Validation.isValidDeleteTask(inputWords)) {
+            DukeException.invalidDeleteTaskException();
+            return;
+        }
+        int taskIndex = Integer.parseInt(inputWords[1]) - 1;
+        if (!Validation.isValidTaskIndex(taskIndex, taskCount)) {
+            DukeException.invalidTaskIndexException();
+            return;
+        }
+        Task current = tasks.get(taskIndex);
+        taskCount = taskCount - 1;
+        UI.printDeleteMessage(current, taskCount);
+        tasks.remove(taskIndex);
+    }
+
+    public static void addTask(String input, String[] inputWords, String command) {
+        String description;
+        switch (command) {
+        case "todo":
+            if (!Validation.isValidTodo(inputWords)) {
+                DukeException.invalidTodoException();
+                return;
+            }
+            description = input.substring(TODO_DESCRIPTION_BEGIN_INDEX);
+            Task newToDo = new ToDo(description);
+            tasks.add(newToDo);
+            break;
+        case "deadline":
+            if (!Validation.isValidDeadline(input, inputWords)) {
+                DukeException.invalidDeadlineException();
+                return;
+            } else if (!Validation.isValidEndDate(input)) {
+                DukeException.invalidEndDateException();
+                return;
+            }
+            String endDate;
+            description = input.substring(DEADLINE_DESCRIPTION_BEGIN_INDEX, input.indexOf("/by") - 1);
+            int endDateBeginIndex = input.indexOf("/by") + 4;
+            endDate = input.substring(endDateBeginIndex);
+            Task newDeadline = new Deadline(description, endDate);
+            tasks.add(newDeadline);
+            break;
+        case "event":
+            if (!Validation.isValidEvent(input, inputWords)) {
+                DukeException.invalidEventException();
+                return;
+            } else if (!Validation.isValidDuration(input)) {
+                DukeException.invalidDurationException();
+                return;
+            }
+            String duration;
+            description = input.substring(EVENT_DESCRIPTION_BEGIN_INDEX, input.indexOf("/at") - 1);
+            int durationBeginIndex = input.indexOf("/at") + 4;
```

Comment:
Avoid the use of magic strings. In this case, the constant string ("/at") can be extracted into a variable with a suitable name.